packages/effect/src/internal/core-effect.ts

```diff
@@ -948,6 +948,20 @@ export const logAnnotations: Effect.Effect<HashMap.HashMap<string, unknown>> = c
     core.currentLogAnnotations
   )
 
+/** @internal */
+export const whenLogLevel = dual<
+  (level: LogLevel.LogLevel) => <A, E, R>(effect: Effect.Effect<A, E, R>) => Effect.Effect<Option.Option<A>, E, R>,
```

Comment:
I think we should support passing a string literal too.

Reply:
added in [1dce660](https://github.com/Effect-TS/effect/pull/4342/commits/1dce6600254f45c64ae02f018aa704622dc416c4)